packages/filebrowser-extension/src/index.ts

```diff
@@ -268,29 +267,17 @@ async function activateFactory(
     const widget = new FileBrowser({ id, model, restore, translator });
 
     // Add a launcher toolbar item.
-    if (labShell) {
```

Comment:
I wonder if removal of this condition might have an impact on @jtpio's jupyterlab-classic.

Reply:
Looks like it should be fine since the `ILabShell` dependency is removed from the plugin.

Reply:
He was the one that made it optional, so I don't think it will impact it at this point (i.e., he already took care of that impact).

Reply:
Sorry, I meant "would not it add the ToollbarButton every to classic which (without checking the code nor making an experiment) currently seems to takes advantage of this if for not having it:

![Screenshot from 2021-01-22 19-45-21](https://user-images.githubusercontent.com/5832902/105537880-821d6800-5cea-11eb-9274-5e5abe46a18e.png)

Reply:
And just prepends custom buttons here: https://github.com/jtpio/jupyterlab-classic/blob/6c26a2652f58e5f05c4f1db79f6e486e5a63d284/packages/tree-extension/src/index.ts#L131-L171

Reply:
Thanks @krassowski. Maybe the toolbar button should be added only if the `launcher:create` command exists then?

Reply:
I thought about that, and moving the launcher:create command check down to the private function. The problem is that depending on the command existing is something that needs to happen at run time. Unlike dependencies on other extensions, where the plugin activation system takes care of ordering the activation of plugins, we don't have the same guarantees with the command system. For example, the launcher extension might add that command after we have loaded, and then we miss adding the button.

We could depend on the launcher extension, but that is probably a backwards dependency as well, and it means that anyone adding a launcher has to then provide that token, rather than just registering a command.

Another way to do it is to make the filebrowser toolbar an open api, and the launcher is the one that adds the button, but I'm not sure that is better either.

Or we could make that launcher button a separate plugin in the filebrowser extension, and Jeremy can just disable that plugin.

In the end, I scaled this PR back down to just fixing the regression, so it is less risky to put in the next patch release.

Reply:
Assuming there are no concurrency issues with command registration then it sounds like a good idea (why should there be a button if the `launcher` is not available).

An alternative could be having an explicit option for disabling this button (which would break your extension if not updated together with lab), but would also work well in a hypothetical case of another extension trying to reuse the filebrowser component in a file-selection dialog (not saying that it is a good idea); having a `create folder`, `upload` and `refresh` buttons would still make sense there, but having `New launcher` button not so much. Obviously one could just use the filebrowser and not filebrowser-extension.

Reply:
Can I suggest that we take this design conversation to a new issue, and focus this issue on solving the problem @krassowski brought up about opening a new launcher in a new browser tab in simple interface?

Reply:
> Or we could make that launcher button a separate plugin in the filebrowser extension

This sounds like something that could be indeed be done in a separate PR.

For reference there is also https://github.com/jupyterlab/jupyterlab/pull/9667 that tries to make the file browser more flexible and reusable in other lab-based apps.